packages/neutrino/bin/neutrino

```diff
@@ -53,26 +50,58 @@ const args = yargs
   .help()
   .argv;
 
-function run(command, args) {
-  const pkg = optional(join(process.cwd(), 'package.json')) || {};
+// getPackageJson :: () -> Object
+const getPackageJson = memoize(() => optional(join(process.cwd(), 'package.json')) || {});
+
+// foldMiddleware :: Object -> Array
+const foldMiddleware = (args) => {
+  const pkg = getPackageJson();
   const pkgMiddleware = pathOr([], ['neutrino', 'use'], pkg);
-  const middleware = [...new Set(pkgMiddleware.concat(args.use))];
+
+  return [...new Set(pkgMiddleware.concat(args.use))];
+};
+
+// normalizeOptions :: Object -> Object
+const normalizeOptions = (args, command) => {
+  const pkg = getPackageJson();
   const options = pathOr({}, ['neutrino', 'options'], pkg);
   const config = pathOr({}, ['neutrino', 'config'], pkg);
-  const api = new Neutrino(Object.assign(options, { config }));
+  const args = Object.assign(args, { env: command === 'inspect' ? getNodeEnv(command) : args.env });
 
-  // Grab all middleware and merge them into a single webpack-chain config instance
-  api.import(middleware);
+  return Object.assign(options, { config, args });
+};
 
-  // Also grab any Neutrino config from package.json and merge it into the config at a higher precedence
-  api.use(() => api.config.merge(config));
+// fail :: Error|Array Error -> IO ()
+const fail = (err) => {
+  const errors = Array.isArray(err) ? err : [err];
```

Comment:
Maybe we could add a `castToArray` method to utils. I believe this is used in multiple places.